File: medistock-backend/src/main/java/com/medistock/medistock_backend/service/UserService.java
package com.medistock.medistock_backend.service;

import com.medistock.medistock_backend.dto.UserDto;
import com.medistock.medistock_backend.dto.UserRequest;

import java.util.List;

public interface UserService {
    List<UserDto> getAllUsers();
    UserDto getUserById(Long id);
    UserDto createUser(UserRequest userRequest);
    UserDto updateUser(Long id, UserRequest userRequest);
    void deleteUser(Long id);
}

File: frontend/src/services/api/userService.js
import apiClient from './apiClient';

export const userService = {
  getAllUsers: async () => {
    const response = await apiClient.get('/admin/users');
    return response.data;
  },

  getUserById: async (id) => {
    const response = await apiClient.get(`/admin/users/${id}`);
    return response.data;
  },

  updateUser: async (id, data) => {
    const response = await apiClient.put(`/admin/users/${id}`, data);
    return response.data;
  },

  deleteUser: async (id) => {
    const response = await apiClient.delete(`/admin/users/${id}`);
    return response.data;
  },
};
